benchmarks/test_nonlimber.py

```diff
@@ -0,0 +1,269 @@
+            dndz=(Nzs["z_cl"], Nz),
+            bias=(Nzs["z_cl"], np.ones(len(Nzs["z_cl"]))))
+        t_g.append(t)
+    t_s = []
+    for Nz in Nzs["dNdz_sh"].T:
+        t = ccl.WeakLensingTracer(cosmo, dndz=(Nzs["z_sh"], Nz))
+        t_s.append(t)
+    ells = get_ells()
+    raw_truth = read_cls()
+    indices_gg = []
+    indices_gs = []
+    indices_ss = []
+    rind_gg = {}
+    rind_gs = {}
+    rind_ss = {}
+    Ng, Ns = len(t_g), len(t_s)
+    for i1 in range(Ng):
+        for i2 in range(i1, Ng):
+            rind_gg[(i1, i2)] = len(indices_gg)
+            rind_gg[(i2, i1)] = len(indices_gg)
+            indices_gg.append((i1, i2))
+
+        for i2 in range(Ns):
+            rind_gs[(i1, i2)] = len(indices_gs)
+            rind_gs[(i2, i1)] = len(indices_gs)
+            indices_gs.append((i1, i2))
+
+    for i1 in range(Ns):
+        for i2 in range(i1, Ns):
+            rind_ss[(i1, i2)] = len(indices_ss)
+            rind_ss[(i2, i1)] = len(indices_ss)
+            indices_ss.append((i1, i2))
+
+    # Sanity checks
+    assert np.allclose(raw_truth[0], ells)
+    Nell = len(ells)
+    tgg, tgs, tss = raw_truth[1:]
+    assert tgg.shape == (len(indices_gg), Nell)
+    assert tgs.shape == (len(indices_gs), Nell)
+    assert tss.shape == (len(indices_ss), Nell)
+
+    # now generate errors
+    err_gg = []
+    err_gs = []
+    err_ss = []
+    nmodes = get_nmodes()
+    for i1, i2 in indices_gg:
+        err_gg.append(
+            np.sqrt(
+                (
+                    tgg[rind_gg[(i1, i1)]] * tgg[rind_gg[(i2, i2)]]
+                    + tgg[rind_gg[(i1, i2)]] ** 2
+                )
+                / nmodes
+            )
+        )
+    for i1, i2 in indices_gs:
+        err_gs.append(
+            np.sqrt(
+                (
+                    tgg[rind_gg[(i1, i1)]] * tss[rind_ss[(i2, i2)]]
+                    + tgs[rind_gs[(i1, i2)]] ** 2
+                )
+                / nmodes
+            )
+        )
+    for i1, i2 in indices_ss:
+        err_ss.append(
+            np.sqrt(
+                (
+                    tss[rind_ss[(i1, i1)]] * tss[rind_ss[(i2, i2)]]
+                    + tss[rind_ss[(i1, i2)]] ** 2
+                )
+                / nmodes
+            )
+        )
+    # Finally get the PT thing going
+    ptc = pt.EulerianPTCalculator(with_NC=True, with_IA=False,
+                                  log10k_min=-4, log10k_max=4,
+                                  nk_per_decade=200)
+    ptc.update_ingredients(cosmo)
+    ptc_lin = pt.EulerianPTCalculator(with_NC=True, with_IA=False,
+                                      log10k_min=-4, log10k_max=4,
+                                      nk_per_decade=200,
+                                      b1_pk_kind='linear',
+                                      bk2_pk_kind='linear')
+    ptc_lin.update_ingredients(cosmo)
+    ptt_g = [pt.PTNumberCountsTracer(b1=bias) for bias in tpar["b_g"]]
+    ptt_m = pt.PTMatterTracer()
```

Comment:
Humm, why do you need any perturbation theory here? You're only testing linearly biased things, right? In which case you can just pass the bias in the `NumberCountsTracer` objects. Perhaps I'm missing something.

Reply:
This test confirms that using PTTracers in the non-limber code still works as-expected.

Reply:
but is it right that you could achieve the same result simply passing biases through `NumberCountsTracer`s and using the linear and nonlinear matter power spectra?

Reply:
I think we should at some point get to the bottom of this. I grudingly added these PTtracer into tests, but fundamentally I still don't understand why does the non-limber calculate ever needs to know the linear theory.

Reply:
This is possible for NZTracers, but not PTTracers. PTTracers can only assume _one_ underlying pk model. This code is as-written for generality regardless of tracer type.